File: pad.py
import os

dir_path = "app/(consumer)"
files = [f for f in os.listdir(dir_path) if f.endswith(".tsx")]

swaps = 0
for f in files:
    if f in ['profile.tsx', 'edit-profile.tsx', 'settings.tsx', 'home.tsx']:
        continue
        
    full_path = os.path.join(dir_path, f)
    with open(full_path, "r") as fp:
        lines = fp.readlines()
        
    content = "".join(lines)
    if "<ScreenScaffold" in content:
        continue
    if "paddingTop: Platform.OS === 'web'" in content:
        continue
        
    modified = False
    for i, line in enumerate(lines):
        if "paddingHorizontal" in line and "isDesktopLayout" in line:
            # check surroundings for paddingTop
            has_padding_top = False
            for j in range(max(0, i-3), min(len(lines), i+4)):
                if "paddingTop" in lines[j]:
                    has_padding_top = True
                    if "Platform.OS" not in lines[j] and "isDesktopLayout ? 24 : insets.top" in lines[j]:
                        lines[j] = lines[j].replace("isDesktopLayout ? 24", "Platform.OS === 'web' ? 80")
                        modified = True
            
            if not has_padding_top:
                # Add paddingTop
                lines.insert(i+1, "                        paddingTop: Platform.OS === 'web' ? 80 : undefined,\n")
                modified = True
                break
                
    if modified:
        with open(full_path, "w") as fp:
            fp.writelines(lines)
        print("Modified", f)
        swaps += 1

print(f"Total modified: {swaps}")

File: pad.js
const fs = require('fs');
const path = require('path');

const dir = path.join(__dirname, 'app/(consumer)');
const files = fs.readdirSync(dir).filter(f => f.endsWith('.tsx'));

for (const file of files) {
    if (file === 'profile.tsx' || file === 'edit-profile.tsx' || file === 'settings.tsx' || file === 'home.tsx') continue; // already modified or uses ScreenScaffold without need

    const fullPath = path.join(dir, file);
    let content = fs.readFileSync(fullPath, 'utf8');

    // Quick regex to find contentContainerStyle inside ScrollView or FlatList that may need paddingTop
    // We want to add paddingTop: Platform.OS === 'web' ? 80 : insets.top + xx
    
    // Check if file uses ScreenScaffold
    if (content.includes('<ScreenScaffold')) {
        // If it includes ScreenScaffold, the scaffold handles padding, BUT wait.
        // Some files pass styling explicitly to FlatList even with ScreenScaffold (like marketplace.tsx and collection.tsx)
        // If so, they usually already have paddingTop.
        continue;
    }

    if (content.includes('paddingTop: Platform.OS === \'web\'')) {
        continue;
    }

    let modified = false;

    // We look for typical padding declarations to inject ours.
    // Replace paddingHorizontal: isDesktopLayout ? 32 : 16,
    // with paddingHorizontal: ..., paddingTop: Platform.OS === 'web' ? 80 : undefined,
    // (Only if it doesn't already have paddingTop right after or before)
    
    const lines = content.split('\n');
    for (let i = 0; i < lines.length; i++) {
        if (lines[i].includes('paddingHorizontal') && lines[i].includes('isDesktopLayout')) {
            // Check surrounding lines for paddingTop
            let hasPaddingTop = false;
            for(let j = Math.max(0, i-3); j < Math.min(lines.length, i+4); j++) {
                if (lines[j].includes('paddingTop')) {
                    hasPaddingTop = true;
                    // Fix existing paddingTop if it doesn't account for web
                    if (!lines[j].includes('Platform.OS') && lines[j].includes('isDesktopLayout ? 24 : insets.top')) {
                        lines[j] = lines[j].replace('isDesktopLayout ? 24', 'Platform.OS === \'web\' ? 80');
                        modified = true;
                    } else if (!lines[j].includes('Platform.OS') && lines[j].includes('insets.top')) {
                       // Probably mobile specific, we can prepend `Platform.OS === 'web' ? 80 : `
                       // But let's be careful.
                    }
                }
            }
            if (!hasPaddingTop) {
                // Insert it after this line
                lines.splice(i+1, 0, `                        paddingTop: Platform.OS === 'web' ? 80 : undefined,`);
                modified = true;
                break;
            }
        }
    }
    
    if (modified) {
        fs.writeFileSync(fullPath, lines.join('\n'));
        console.log('Modified', file);
    }
}
